src/bitcoin-cli.cpp

```diff
@@ -85,7 +85,7 @@ static bool AppInitRPC(int argc, char* argv[])
         }
 
         fprintf(stdout, "%s", strUsage.c_str());
-        return false;
+        if(argc < 2) return false; else exit(0);
```

Comment:
I don't think the code is very readable this way, there is nothing that uses `-1` as a success value.

You could improve it by defining a constant for success value (that happens to be -1, or whatever else), and document that the function returns either a process return code or this magic return success code.